e2e/native-esm/__tests__/native-esm.test.js

```diff
@@ -244,6 +244,56 @@ test('can mock transitive module', async () => {
   expect(importedMock.foo).toBe('bar');
 });
 
+test('can unmock module', async () => {
+  jestObject.unstable_mockModule('../index.js', () => ({
+    double: num => num * 1000,
+  }));
+
+  const importedMock = await import('../index.js');
+  expect(importedMock.double(2)).toBe(2000);
+
+  jestObject.unstable_unmockModule('../index.js');
+
+  const importedMockAfterUnmock = await import('../index.js');
+  expect(importedMockAfterUnmock.double(2)).toBe(4);
+});
+
+test('can unmock transitive module', async () => {
+  jestObject.unstable_mockModule('../index.js', () => ({
+    double: num => num * 1000,
+  }));
+
+  const importedMock = await import('../reexport.js');
+  expect(importedMock.double(2)).toBe(2000);
+
+  jestObject.unstable_unmockModule('../index.js');
+
+  const importedMockAfterUnmock = await import('../reexport.js');
```

Comment:
I don't think this will work as `../reexport.js` already has a reference to the old (mocked) `index.js`.

Do you have an example of a test of ours using `unmock` that works in this way? If not, I think you can just delete the test

Reply:
Yes. It is not work with "mock" and "unmock"
```
test('test', () => {
    jest.mock('./dependencyModule', () => {
        return {getSecretNumber: () => 1000};
    });

    const {getSecretNumber} = require("./dependencyModule");
    console.log(getSecretNumber()); // output original implementation. Not mock impl.

    jest.unmock('./dependencyModule'); // it is the reset implementation in the file
});
```